### PR TITLE
AtlasEngine: Recognize Powerline glyphs

### LINKED ISSUE #13029: Powerline glyph does not fill whole 'cell' / 'box' height
### Windows Terminal version

1.12.10982.0

### Windows build number

10.0.19044.1645

### Other Software

Cascadia Code 2111.01

### Steps to reproduce

* Install static `Cascadia Code PL Regular` [1]:
  * `ttf/static/CascadiaCodePL-Regular.ttf` or
  * `otf/static/CascadiaCodePL-Regular.otf`
* Use it as font with 12 pt
* Utilize the powerline glyphs like so (copy and paste into command prompt terminal window):
  `set prompt=$E[0;33;40m$E[7;33;40m Manuel $E[0;33;40m$E[7;32;40m  ~ $E[0;32;40m$E[7;34;40m  $E[0;34;40m$E[0;37;40m $g`

[1] https://github.com/microsoft/cascadia-code/releases/tag/v2111.01

### Expected Behavior

The upper and lower 'border' of the powerline glyphs shall line up with the 'cell' top and bottom.
Cell top and bottom can be seen if insert a blank with reverse colors (like done in the prompt command above).

This is how it should look like, roughly (magnified 400%):

image

### Actual Behavior

The glyphs are about 1 pixel to small vertically; here with the `otf` one pixel at the bottom seems to be missing:

image

This is most notable with the rounded D shape powerline symbols, but in fact are present in all.

image

The static `ttf` seem to have the missing pixel in the top.
I did not try the VF.

Origin of issue: Bottom (after closing) https://github.com/ryanoasis/nerd-fonts/issues/828

I believe this is not a Cascadia Code issue, because the original issue came up with a Nerd Font patched version of that font (Caskaydia Cove); and in the process of that issue I tried a lot like ascent and descent changing (all of them OS/2, typo, Win etc ;), PPEM modification, etc pp, all to no avail.
I do not really know how `Windows Terminal` finds out its cell size. If one understands that maybe the font can be beat into function. But I fear that it is a rendering issue, as this is always 1 px big, regardless of font size in point.

_Edit: Fix typo_

### PR DESCRIPTION
**Pull request by lhecker**
This commit makes AtlasEngine recognize Powerline glyphs as box drawing ones.
The extra pixel offsets when determining the `scale` caused weird artifacts
and thus were removed. It seems like this causes no noticeable regressions.

Closes #13029

## Validation Steps Performed
* Run all values of `wchar_t` through `isInInversionList`
  and ensure it produces the expected value ✅
* Powerline glyphs are correctly scaled with Cascadia Code PL ✅

### CODE REVIEW

**Comment by lhecker** [comment]
Before:
image

After:
image

**Review by carlos-zamora — APPROVED** [review]
TIL inversion lists. Cool!

**Review by zadjii-msft — APPROVED** [review]
I love this and feel like _why haven't we used this more before_

File: src/renderer/atlas/AtlasEngine.r.cpp [hunk 1]
```diff
@@ -329,8 +367,8 @@ void AtlasEngine::_drawGlyph(const AtlasQueueItem& item) const
             // So for safe measure we'll always scale them to the exact size.
             // But add 1px to the destination size, so that we don't end up with fractional pixels.
             scalingRequired = true;
-            scale.x = (layoutBox.x + _r.dipPerPixel) / boxSize.x;
-            scale.y = (layoutBox.y + _r.dipPerPixel) / boxSize.y;
+            scale.x = layoutBox.x / boxSize.x;
```
Comment (DHowett) [inline src/renderer/atlas/AtlasEngine.r.cpp:370]: wat, why'd this need to change again?

**Review by DHowett — COMMENTED** [review]
My only concern with this is .. two concerns.

Powerline is more standardized, ironically, than block drawing characters. We'll want to eventually offer an integrated PL glyph block if we can.

Also, Powerline occupies the Private Use Area. This will cause rendering issues for anybody else who uses the PUA, right?

File: src/renderer/atlas/AtlasEngine.r.cpp (hunk 1 above)
Comment (lhecker) [reply src/renderer/atlas/AtlasEngine.r.cpp:370]: The extra pixel added to the `layoutBox` causes the vertical lines here:
image

Without the extra pixel, that issue doesn't occur even if we don't recognize the glyph as a box drawing one.

**Comment by lhecker** [comment]
> Powerline is more standardized, ironically, than block drawing characters. We'll want to eventually offer an integrated PL glyph block if we can.

Agreed!

> Also, Powerline occupies the Private Use Area. This will cause rendering issues for anybody else who uses the PUA, right?

I think that's possible, but I believe it's unlikely and so I'd like to wait and see whether anyone complains first. Basically there's evidence that we should recognize them, but no evidence yet that we should be careful about it.

**Pull request merged**